ansible_collections/arista/avd/roles/eos_designs/python_modules/ip_addressing/avdipaddressing.py

```diff
@@ -35,6 +35,33 @@ def _template(self, template_path, **kwargs):
         template_vars = ChainMap(kwargs, self._hostvars)
         return self.shared_utils.template_var(template_path, template_vars)
 
+    def _mlag_ip(self, pool: str, ip_offset: int) -> str:
+        compact_schema = self._mlag_compact_addressing
+
+        if compact_schema:
+            offset = self._mlag_odd_id_based_offset()
+            return self._ip(pool, 31, offset, ip_offset, False)
+        
+        offset = self._mlag_primary_id - 1
+        return self._ip(pool, 31, offset, ip_offset, True)
+
+    def _mlag_odd_id_based_offset(self) -> int:
+        """
+        Return the subnet offset for an MLAG pair based on odd id
+
+        Requires a pair of odd and even IDs
+        """
+
+        # Verify a mix of odd and even IDs
+        if (self._mlag_primary_id % 2) + (self._mlag_secondary_id % 2) != 1:
+            raise AristaAvdError(f"MLAG compact addressing mode requires all MLAG pairs to have a single odd and even ID")
+
+        odd_id = self._mlag_primary_id
+        if odd_id % 2 == 0:
+            odd_id = self._mlag_secondary_id
+        
+        return odd_id - 1
```

Comment:
Instead of your change to the base _ip function, you should `return (odd_id - 1) / 2`.
So:
ID: 1 -> offset 0
ID: 3 -> offset 1
etc.

Reply:
That did not solve it when I tried that

Reply:
We should just tell `_ip()` which subnet we want of a certain size. Not hack it to behave differently.

Reply:
Turns out it was a weird behavior with returning a float 1.0 vs int 1